common/debug.c

```diff
@@ -0,0 +1,257 @@
+#include "debug.h"
+
+void bundle_dump(bundle_transactions_t *bundle) {
+  iota_transaction_t *curr_tx = NULL;
+  BUNDLE_FOREACH(bundle, curr_tx) { transaction_obj_dump(curr_tx); }
+}
+
+void transaction_obj_dump(iota_transaction_t *tx_obj) {
+  field_mask_t old_mask = {};
+  memcpy(&old_mask, &tx_obj->loaded_columns_mask, sizeof(field_mask_t));
+  memset(&tx_obj->loaded_columns_mask, 0xFFFFF, sizeof(field_mask_t));
+
+  printf("==========Transaction Object==========\n");
+  // address
+  printf("address: ");
+  flex_trit_print(transaction_address(tx_obj), NUM_TRITS_ADDRESS);
+  printf("\n");
+
+  printf("value: %" PRId64 "\n", transaction_value(tx_obj));
+
+  printf("obsolete tag: ");
+  flex_trit_print(transaction_obsolete_tag(tx_obj), NUM_TRITS_OBSOLETE_TAG);
+  printf("\n");
+
+  printf("timestamp: %" PRId64 "\n", transaction_timestamp(tx_obj));
+  printf("curr index: %" PRId64 " \nlast index: %" PRId64 "\n", transaction_current_index(tx_obj),
+         transaction_last_index(tx_obj));
+
+  printf("bundle: ");
+  flex_trit_print(transaction_bundle(tx_obj), NUM_TRITS_BUNDLE);
+  printf("\n");
+
+  printf("trunk: ");
+  flex_trit_print(transaction_trunk(tx_obj), NUM_TRITS_TRUNK);
+  printf("\n");
+
+  printf("branch: ");
+  flex_trit_print(transaction_branch(tx_obj), NUM_TRITS_BRANCH);
+  printf("\n");
+
+  printf("tag: ");
+  flex_trit_print(transaction_tag(tx_obj), NUM_TRITS_TAG);
+  printf("\n");
+
+  printf("attachment_timestamp: %" PRId64 "\n", transaction_attachment_timestamp(tx_obj));
+  printf("attachment_timestamp_lower: %" PRId64 "\n", transaction_attachment_timestamp_lower(tx_obj));
+  printf("attachment_timestamp_upper: %" PRId64 "\n", transaction_attachment_timestamp_upper(tx_obj));
+
+  printf("nonce: ");
+  flex_trit_print(transaction_nonce(tx_obj), NUM_TRITS_NONCE);
+  printf("\n");
+
+  printf("hash: ");
+  flex_trit_print(transaction_hash(tx_obj), NUM_TRITS_HASH);
+  printf("\n");
+
+  printf("message: \n");
+  flex_trit_print(transaction_message(tx_obj), NUM_TRITS_MESSAGE);
+  printf("\n");
+
+  memcpy(&tx_obj->loaded_columns_mask, &old_mask, sizeof(field_mask_t));
+}
+
+bool transaction_cmp(iota_transaction_t *tx1, iota_transaction_t *tx2) {
```

Comment:
Can this function print log only in debug mode? Therefore, we can use it more widely.

Reply:
compare two transaction object is easier to use `memcmp`